ui/src/message_popup/renderers/instant_audio_recorder.jsx

```diff
@@ -0,0 +1,144 @@
+/* @flow weak */
+import React from 'react';
+import RaisedButton from 'material-ui/RaisedButton';
+import AudioCapture from '../../components/audio_capture.jsx';
+
+type State = {
+  haveRecorded: bool,
+  downloadUrl: ?string,
+  uploadState: 'idle' | 'pending' | 'done',
+  blob: ?Blob,
+  uploadedUrl: ?string
+};
+
+export default React.createClass({
+  displayName: 'InstantAudioRecorder',
+
+  propTypes: {
+    url: React.PropTypes.string.isRequired,
+    onDoneUploading: React.PropTypes.func.isRequired,
+    onLogMessage: React.PropTypes.func.isRequired,
+    instantRecord: React.PropTypes.bool,
+    recordingPrompt: React.PropTypes.string,
+    recordingText: React.PropTypes.string,
+  },
+
+  getDefaultProps() {
+    return {
+      instantRecord: false,
+      recordingPrompt: "Recording...",
+      recordingText: "Done",
+    };
+  },
+
+  getInitialState():State {
+    return {
+      haveRecorded: false,
+      downloadUrl: null,
+      uploadState: 'idle',
+      blob: null,
+      uploadedUrl: null
+    };
+  },
+
+
+  uploadBlob(blob:Blob) {
+    const {url} = this.props;
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', url);
+    xhr.onload = this.onDoneUploading;
+    xhr.onerror = this.onErrorUploading;
+    xhr.send(blob);
+  },
+
+  isRecording() {
+    return this.whichStep(this.state) === 'recording';
+  },
+
+  // Determine which step we're at in the flow through
+  // these states.
+  whichStep(state:State) {
+    const {
+      haveRecorded,
+      blob,
+      uploadState,
+      uploadedUrl
+    } = state;
+
+    if (!this.props.instantRecord && !haveRecorded) return 'idle';
+    if (this.props.instantRecord && !haveRecorded) return 'recording';
+    if (haveRecorded && !blob) return 'processing';
+    if (blob && uploadState === 'pending') return 'saving';
+    if (blob && uploadedUrl) return 'done';
+  },
+
+  onDoneRecordingClicked() {
+    this.props.onLogMessage('message_popup_audio_done_recording_clicked');
```

Comment:
We should chat after the playtest about the message type strings here - right now if they're the same as other components, so in looking at the data on the server there's no way to distinguish between recording that was done with the instant component or the "review" one.  We might want to use different message types per component to capture more of this, but the tradeoff is more message types when doing analysis.